apps/naive/lib/naive/strategy.ex

```diff
@@ -511,10 +508,24 @@ defmodule Naive.Strategy do
     }
   end
 
-  def update_status(symbol, status)
-      when is_binary(symbol) and is_binary(status) do
-    @repo.get_by(Settings, symbol: symbol)
+  def update_status(id, status)
+      when is_binary(status) do
+    @repo.get_by(Traders, id: id)
     |> Ecto.Changeset.change(%{status: status})
     |> @repo.update()
   end
+
+  defp new_order(%Exchange.Order{} = order, trader_id) do
+    order
+    |> Map.put(:trader_id, trader_id)
+    |> broadcast_order()
+  end
+
+  defp broadcast_order(%Exchange.Order{} = order) do
+    @pubsub_client.broadcast(
+      Core.PubSub,
+      "ORDERS:#{order.symbol}",
+      order
+    )
+  end
```

Comment:
Do these belong in this file? It seems that we'd need them for any strategy, not just `naive`

Reply:
I'd say yes, this module is now more responsible for trading infrastructure than any particular strategy. This particular snippet is responsible to ensure that the `trader_id` is passed to the DataWarehouse. Maybe we should make the rename it from `Naive` to `Trader` already.

Reply:
+1 to rename away from `Naive` if the module is not responsible for the `naive` strategy anymore